### PR TITLE
RR-849-add-form-scaffolding

### PR DESCRIPTION
**Pull request by chopkinsmade**
## Description of change

- Added a new page with an empty form to add new export pipeline items, along with the routing needed to access the page
- The same form will be used by both the add and edit pages, so a separate ExportFormFields.jsx component has been created ready

## Test instructions

Use the url `http://localhost:3000/export/create?companyId=c1ea5ee7-c6f5-4dd1-8c5e-785c5a115719` to load the new page (you can use any valid company id). There should be a breadcrumb with the company name, and the same company name as a sub heading.
The form will be empty, but clicking cancel will redirect to the company page that is provided in the `?companyId` query string

To test the error screen, use the url `http://localhost:3000/export/create?companyId=not_real`

### Valid company:
image

### Invalid company
image

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [ ] Has the branch been rebased to main?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: src/client/modules/ExportPipeline/ExportForm/ExportFormAdd.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    >
+      <Task.Status
+        name={TASK_GET_COMPANY_DETAIL}
+        id={COMPANY_DETAILS_ID}
+        progressMessage="Loading company details"
+        startOnRender={{
+          payload: companyId,
+          onSuccessDispatch: COMPANY_LOADED,
+        }}
+        renderError={ErrorHandler}
```
Comment (paulgain) [inline src/client/modules/ExportPipeline/ExportForm/ExportFormAdd.jsx:85]: There's no need to define this on a Task, the component has it baked in.

File: src/client/modules/ExportPipeline/ExportForm/ExportFormFields.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+
+import Form from '../../../components/Form'
+import { FormLayout } from '../../../../client/components'
+import { FORM_LAYOUT } from '../../../../common/constants'
+import urls from '../../../../lib/urls'
+import { TASK_SAVE_EXPORT } from './state'
+
+const ExportFormFields = ({ companyId, analyticsFormName }) => {
+  return (
```
Comment (paulgain) [inline src/client/modules/ExportPipeline/ExportForm/ExportFormFields.jsx:10]: Favour implicit returns over explicit returns:
```js
const ExportFormFields = ({ companyId, analyticsFormName }) => (<>...</>)
```

File: src/client/modules/ExportPipeline/ExportForm/ExportFormAdd.jsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      text: 'Home',
+    },
+    {
+      link: urls.companies.index(),
+      text: 'Companies',
+    },
+    { text: 'Add export' },
+  ]
+
+  if (Object.keys(company).length > 0) {
```
Comment (paulgain) [inline src/client/modules/ExportPipeline/ExportForm/ExportFormAdd.jsx:51]: Can you not just test for the company?
```js
if (company) { ... }
```

**Review by paulgain — COMMENTED** [review]
Looks good, although the breadcrumbs section looks verbose.

File: src/client/modules/ExportPipeline/ExportForm/ExportFormAdd.jsx (hunk 3 above)
Comment (paulgain) [reply src/client/modules/ExportPipeline/ExportForm/ExportFormAdd.jsx:51]: That's the problem, it shouldn't be defaulting to an empty object - see my comment below.

**Review by paulgain — APPROVED** [review]

File: src/client/modules/ExportPipeline/ExportForm/ExportFormAdd.jsx (hunk 1 above)
Comment (chopkinsmade) [reply src/client/modules/ExportPipeline/ExportForm/ExportFormAdd.jsx:84]: I'll merge this in with a custom error handler, as there is a wider conversation to be had about how we handle errors from the API when we aren't going via an express route. Will look at some refactoring for the edit version of this form, as that will also trigger a direct API call avoiding express

**Pull request merged**